### ISSUE TITLE
Draft spec for extended partial methods

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/csharplang#3417]
Related to #3301
Proposal: https://github.com/dotnet/csharplang/blob/master/proposals/csharp-9.0/extending-partial-methods.md

C# 9 introduces a new kind of [partial method](https://github.com/dotnet/csharplang/blob/master/spec/classes.md#partial-methods) called an *extended partial method*. An *extended partial method* must have both a defining declaration and an implementing declaration.

They are required to have explicit accessibility modifiers, which must be identical on both declarations. They are permitted to have `out` parameters and to have non-`void` return types. They are permitted to use the `virtual`, `override`, `sealed`, and `new` modifiers. Both declarations must contain identical combinations of these modifiers, although they may be present in a different order in each declaration.

A single declaration of an *extended partial method* is said to be an *implementing partial declaration* if it has a body or if it has an `extern` modifier. Only the implementing declaration is permitted to have an `extern` modifier.

**Comment by ufcpp** [comment]
Can't explicit implementation of an interface method be partial?

```cs
interface I
{
    int M();
}

partial class C : I
{
    public partial int I.M(); // requires access modifier but explicit implementation bans it
}

partial class C : I
{
    public partial int I.M() { }
}
```

**Comment by HaloFour** [comment]
Maybe the explicit implementation syntax needs to optionally allow a `private` access modifier?

**Comment by RikkiGibson** [comment]
The feature as shipped in VS 16.7-preview2 does not support partial explicit interface implementations. The compiler team may reconsider this limitation based on user feedback.
